### PR TITLE
emul: Implement emul_get_binding

### PR DESCRIPTION
**Pull request by yperess**
Issue #38271

Implement a getter for emulators similar to device_get_binding. This
function can be used to get the emulator instance during tests to call
emulator specific functions.

Example: The current BMI160 emulator pre-defines a finite set of data
samples that will be returned. If a test was to be written for logic
that uses that data, then the emulator would become completely useless
without the ability for the test to define what data should be returned.

This will also help in exercising error conditions in tests.

Signed-off-by: Yuval Peress <peress@chromium.org>

### CODE REVIEW

File: include/drivers/emul.h
Comment (sjg20) [inline include/drivers/emul.h:105]: Could you note where the 'name' comes from? That would be helpful for people I think.

Also for the commit subject, how about 'emul: Support getting an emulator by name?' as emul_get_binding() seems a bit unclear/specific

**Review by sjg20 — APPROVED** [review]

File: include/drivers/emul.h
Comment (yperess) [reply include/drivers/emul.h:107]: Done, I paraphrased the docs from `device_get_binding()` to keep things as consistent as possible.

**Comment by yperess** [comment]
@jackrosenthal Would you mind taking a look at this one? I can't add reviewers and I need a second approval before this can merge.

File: subsys/emul/emul.c
Comment (jackrosenthal) [inline subsys/emul/emul.c:21]: nit: prefer `!strcmp(a, b)` to `strcmp(a, b) == 0`

**Review by jackrosenthal — APPROVED** [review]

**Pull request merged**
